clang/lib/Sema/SemaStmt.cpp

```diff
@@ -3967,6 +3968,15 @@ bool Sema::DeduceFunctionTypeFromReturnExpr(FunctionDecl *FD,
     // Update all declarations of the function to have the deduced return type.
     Context.adjustDeducedFunctionResultType(FD, Deduced);
 
+      // Print the deduced return type
+  SourceLocation Loc = FD->getLocation();
+  if(DumpAutoTypeInference.getNumOccurrences())
```

Comment:
is the check
```
SM.isWrittenInMainFile(Loc)
```
 not required for functions?

Reply:
Added the check